pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -299,33 +280,134 @@ def build_model(
+        likelihood_config = model_config.get("likelihood", {})
+        likelihood_type = likelihood_config.get("type")
+        dims = likelihood_config.get("dims")
+
+        if not likelihood_type:
+            raise ValueError("Likelihood type must be specified in the model config.")
+
+        likelihood_func = getattr(pm, likelihood_type, None)
+        if likelihood_func is None:
+            raise ValueError(f"Invalid likelihood type {likelihood_type}")
+
+        # Transform mu if the likelihood type is Lognormal or HurdleLognormal
+        if likelihood_type in ['LogNormal', 'HurdleLogNormal']:
+            mu = pt.log(mu)
+
+        # Create sub-priors
+        sub_priors = {}
+        for param, config in likelihood_config.items():
+            if param not in ['type', 'dims']:  # Skip 'type' and 'dims'
+                if param == 'params':  # Handle nested 'params'
+                    for sub_param, sub_config in config.items():
+                        sub_priors[sub_param] = self.create_priors_from_config({sub_param: sub_config})[sub_param]
+                else:
+                    sub_priors[param] = self.create_priors_from_config({param: config})[param]
+
+        return likelihood_func(name="likelihood", mu=mu, observed=target_, dims=dims,  **sub_priors)
+
+    def create_tvp_priors(self, param, config, length, positive=False):
+        dims = config.get("dims", None)  # Extracting dims from the config
+        print(dims)
+        gp_list = [self.gp_wrapper(name=f"{param}_{i}", X=np.arange(len(self.X[self.date_column]))[:, None], config=config, positive=positive) for i in range(length)]
+        stacked_gp = pt.stack(gp_list, axis=1)
+        return pm.Deterministic(f"{param}", stacked_gp, dims=dims)
+
+
+    def gp_wrapper(self, name, X, config, positive=False, **kwargs):
+        return self.gp_coeff(X, name, config=config, positive=positive, **kwargs)
+
+
+    def gp_coeff(self, X, name, mean=0.0, positive=False, config=None):
+        params = pm.find_constrained_prior(pm.InverseGamma, 0.5, 2, init_guess={"alpha": 2, "beta": 1}, mass=0.95)
+        ell = pm.InverseGamma(f"ell_{name}", **params)
+        eta = pm.Exponential(f"_eta_{name}", lam=1 / 0.5)
+        cov = eta ** 2 * pm.gp.cov.ExpQuad(1, ls=ell)
+    
+        gp = pm.gp.HSGP(m=[20], c=1.3, cov_func=cov)
+        f_raw = gp.prior(f"{name}_tvp_raw", X=X)
+    
+
+        # Inside your gp_coeff function
+        # Offset
+        offset_config = config.get('offset', None) if config else None
+        if offset_config:
+            offset_type = offset_config.get('type')
+            offset_params = {k: v for k, v in offset_config.items() if k != 'type'}
+            offset_prior = getattr(pm, offset_type)(name=f"{name}_offset", **offset_params)
+        else: 
+            offset_prior = 0
+
+        if positive:
+            f_output = pm.Deterministic(f"{name}", (pt.exp(f_raw) - 1) + offset_prior, dims=("date"))
```

Comment:
what is -1?

Reply:
Typo, Bill has reviewed and will change this